eng/scripts/TypeSpec-Compare-CurrentToCodegeneration.ps1

```diff
@@ -51,7 +51,7 @@ if ($LastExitCode -ne 0) {
 The following files are out of date:
 $status
 "
-  exit $LASTEXITCODE
+  exit 1
```

Comment:
`LASTEXITCODE` is code from the `Write-Host`, which is always 0.

This causes the script always return success, even if there is diff. And then error could happen (or not, depends on the diff) in later steps. e.g. https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2937216&view=logs&jobId=b70e5e73-bbb6-5567-0939-8415943fadb9&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=d31d5e2d-d8df-53bc-da94-87e19f5b2ee4

Reply:
It isn't the return code from Write-Host as cmdlets don't set it but it is the return code of the git status call. So if we want to return the same code we need to capture it before the git status call but I think returning 1 here is probably fine as well.

Reply:
```
$ git asdfasdfasdfas
git: 'asdfasdfasdfas' is not a git command. See 'git --help'.
$ $LASTEXITCODE
1
$ write-host foo
foo
$ $LASTEXITCODE
1
```

Reply:
Thanks Wes and Ben! Learnt it.